ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoDivulgenceSpec.scala

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.dao
+
+import java.time.Instant
+import java.util.UUID
+
+import com.daml.lf.data.{ImmArray, Ref}
+import com.daml.lf.transaction.Node.{KeyWithMaintainers, NodeCreate, NodeExercises, NodeFetch}
+import com.daml.lf.transaction.test.TransactionBuilder
+import com.daml.lf.transaction.{Transaction => Tx}
+import com.daml.lf.value.Value.{ContractInst, ValueParty, VersionedValue}
+import com.daml.lf.value.ValueVersion
+import com.daml.platform.store.entries.LedgerEntry
+import org.scalatest.{AsyncFlatSpec, Inside, LoneElement, Matchers}
+
+private[dao] trait JdbcLedgerDaoDivulgenceSpec extends LoneElement with Inside {
+  this: AsyncFlatSpec with Matchers with JdbcLedgerDaoSuite =>
+
+  behavior of "JdbcLedgerDao (divulgence)"
+
+  it should "preserve divulged contracts" in {
+    val (create1, tx1) = {
+      val builder = new TransactionBuilder
+      val contractId = builder.newCid
+      builder.add(
+        NodeCreate(
+          coid = contractId,
+          coinst = someContractInstance,
+          optLocation = None,
+          signatories = Set(alice),
+          stakeholders = Set(alice),
+          key = None
+        )
+      )
+      contractId -> Tx.CommittedTransaction(builder.build())
+    }
+    val (create2, tx2) = {
+      val builder = new TransactionBuilder
+      val contractId = builder.newCid
+      builder.add(
+        NodeCreate(
+          coid = contractId,
+          coinst = someContractInstance,
+          optLocation = None,
+          signatories = Set(bob),
+          stakeholders = Set(bob),
+          key = Some(
+            KeyWithMaintainers(ValueParty(bob), Set(bob))
+          )
+        )
+      )
+      contractId -> Tx.CommittedTransaction(builder.build())
+    }
+    val tx3 = {
+      val builder = new TransactionBuilder
+      val rootExercise = builder.add(
+        NodeExercises(
+          targetCoid = create1,
+          templateId = someTemplateId,
+          choiceId = Ref.ChoiceName.assertFromString("SomeChoice"),
+          optLocation = None,
+          consuming = true,
+          actingParties = Set(bob),
+          chosenValue = someValueRecord,
+          stakeholders = Set(alice, bob),
+          signatories = Set(alice),
+          children = ImmArray.empty,
+          exerciseResult = None,
+          key = None,
+        )
+      )
+      builder.add(
+        NodeFetch(
+          coid = create2,
+          templateId = someTemplateId,
+          optLocation = None,
+          actingParties = Some(Set(bob)),
+          signatories = Set(bob),
+          stakeholders = Set(bob),
+          key = Some(
+            KeyWithMaintainers(ValueParty(bob), Set(bob))
+          ),
+        ),
+        parent = rootExercise,
+      )
+      val nestedExercise = builder.add(
```

Comment:
Is `nestedExercise` missing the parent assignment?

Reply:
Addressed by https://github.com/digital-asset/daml/pull/6607/commits/e64699148e955152ef4f31e54b1cb964b1c50fea